### ISSUE TITLE
Missing nullability warning after assignment to variable in finally block

### ISSUE THREAD

**Issue by gafter** [dotnet/roslyn#33446]
The following should, but does not, produce a warning.
``` c#
class Node
{
    public Node? Next;
    static void Mout(out Node node) => node = null!;
    static void M(Node node)
    {
        try
        {
            Mout(out node.Next);
        }
        finally
        {
            Mout(out node); // might set node to one in which node.Next == null
        }
        node.Next.ToString(); // should be warning
    }
}
```
